csharp/Test/Deployment/NugetApplicationTest.csproj

```diff
@@ -0,0 +1,57 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <IsPackable>true</IsPackable>
+    <PackageId>NugetApplicationTest</PackageId>
+    <Title>NugetApplicationTest</Title>
+    <Description>
+      This test checks that our TypeDB Driver Nuget package works correctly will all the transitive dependencies!
+    </Description>
+    <Authors>TypeDB</Authors>
+    <Company>TypeDB</Company>
+    <Copyright>Copyright © 2024 Vaticle</Copyright>
+    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
+    <PackageLicenseUrl></PackageLicenseUrl>
+    <PackageProjectUrl>https://typedb.com</PackageProjectUrl>
+    <PackageTags>Database TypeDB TypeQL Test</PackageTags>
+    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
+    <RepositoryUrl>https://github.com/vaticle/typedb-driver</RepositoryUrl>
+    <Version>1.0.0</Version>
+    <PackageReleaseNotes>No</PackageReleaseNotes>
+    <GenerateDocumentationFile>false</GenerateDocumentationFile>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <RestoreSources>$(RestoreSources);https://nuget.cloudsmith.io/typedb/public-snapshot/v3/index.json;https://api.nuget.org/v3/index.json</RestoreSources>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="TypeDB.Driver" Version="[DRIVER_CSHARP_VERSION_MARKER]" />
+    <PackageReference Include="TypeDB.Driver.Pinvoke.osx-x64" Version="[DRIVER_CSHARP_VERSION_MARKER]" />
+    <PackageReference Include="TypeDB.Driver.Pinvoke.linux-x64" Version="[DRIVER_CSHARP_VERSION_MARKER]" />
+    <PackageReference Include="TypeDB.Driver.Pinvoke.win-x64" Version="[DRIVER_CSHARP_VERSION_MARKER]" />
+    <PackageReference Include="TypeDB.Driver.Pinvoke.osx-arm64" Version="[DRIVER_CSHARP_VERSION_MARKER]" />
+    <PackageReference Include="TypeDB.Driver.Pinvoke.linux-arm64" Version="[DRIVER_CSHARP_VERSION_MARKER]" />
```

Comment:
Nice this is templated, so we'll know automatically for new versions.

Reply:
It is a test, we substitute these values with a '0.0.0-gitcommitsha' value, which is generated for snapshots.